apps/prairielearn/src/pages/news_item/news_item.js

```diff
@@ -5,61 +5,68 @@ import * as path from 'node:path';
 import { Router } from 'express';
 import asyncHandler from 'express-async-handler';
 
-import * as sqldb from '@prairielearn/postgres';
+import { HttpStatusError } from '@prairielearn/error';
+import { loadSqlEquiv, queryOptionalRow } from '@prairielearn/postgres';
 
-const sql = sqldb.loadSqlEquiv(import.meta.url);
+import { NewsItemSchema } from '../../lib/db-types.js';
+
+import { NewsItem, NewsItemRowSchema } from './news_item.html.js';
+
+const sql = loadSqlEquiv(import.meta.url);
 const router = Router();
 
 router.get(
   '/:news_item_id(\\d+)',
   asyncHandler(async (req, res) => {
-    const result = await sqldb.queryZeroOrOneRowAsync(sql.select_news_item_for_read, {
-      news_item_id: req.params.news_item_id,
-      user_id: res.locals.authn_user.user_id,
-      course_instance_id: res.locals.course_instance ? res.locals.course_instance.id : null,
-      course_id: res.locals.course ? res.locals.course.id : null,
-    });
-    if (result.rowCount === 0) {
-      throw new Error(`invalid news_item_id: ${req.params.news_item_id}`);
+    const newsItem = await queryOptionalRow(
+      sql.select_news_item_for_read,
+      {
+        news_item_id: req.params.news_item_id,
+        user_id: res.locals.authn_user.user_id,
+        course_instance_id: res.locals.course_instance?.id,
+        course_id: res.locals.course?.id,
+      },
+      NewsItemRowSchema,
+    );
+    if (newsItem == null) {
+      throw new HttpStatusError(404, 'Invalid news item ID');
```

Comment:
Minor change in behaviour: this now throws a 404 instead of a generic error. Same for the file route below.

Reply:
Maybe just use a generic "Not Found" here?

Reply:
done